### PR TITLE
Improve expose of update sensors to HA

### PR DESCRIPTION
**Pull request by frenck**
This changes the way update sensors are exposed to Home Assistant.

- Set the entities to have the entity category `diagnostic`
- Re-instate the binary sensor version as default: Home Assistant actually now provides update device classes for binary sensors to indicate an update is pending. This also works with e.g., device automations.
- Sets the device class `update` on the binary sensor

### CODE REVIEW

File: lib/extension/homeassistant.ts [hunk 1]
```diff
@@ -899,24 +899,25 @@ export default class HomeAssistant extends Extension {
...
-                    discovery_payload: {
-                        payload_on: true,
-                        payload_off: false,
-                        value_template: '{{ value_json.update_available}}',
-                        enabled_by_default: false,
-                    },
-                };
-                configs.push(updateAvailableSensor);
-            }
+            const updateAvailableSensor = {
```
Comment (Koenkk) [inline lib/extension/homeassistant.ts:907]: @frenck I see you moved it out of `if (this.legacyApi) {`, was this done intentionally? `update_available` is deprecated and replace by `update.state`.

**Comment by frenck** [comment]
@Koenkk Yes, see OP:

> Re-instate the binary sensor version as default: Home Assistant actually now provides update device classes for binary sensors to indicate an update is pending. This also works with e.g., device automations.

File: lib/extension/homeassistant.ts (hunk 1 above)
Comment (frenck) [reply lib/extension/homeassistant.ts:907]: Yes, see OP:

> Re-instate the binary sensor version as default: Home Assistant actually now provides update device classes for binary sensors to indicate an update is pending. This also works with e.g., device automations.

Comment (frenck) [reply lib/extension/homeassistant.ts:907]: Yes, see OP:

> Re-instate the binary sensor version as default: Home Assistant actually now provides update device classes for binary sensors to indicate an update is pending. This also works with e.g., device automations.

Comment (Koenkk) [reply lib/extension/homeassistant.ts:907]: Fine by me, but we should determine the value based on `update.state` and not `update_available`. Therefore I suggest changing `value_template: '{{ value_json.update_available}}',` to `value_template: '{{ value_json['update']['state'] == "available" }}',`

Comment (frenck) [reply lib/extension/homeassistant.ts:907]: Sounds good to me too.
I'll adjust that soon™

Comment (Koenkk) [reply lib/extension/homeassistant.ts:907]: I will do it, thanks as always!

Comment (frenck) [reply lib/extension/homeassistant.ts:907]: LOL or you do it... fine too 🤣

**Pull request merged**